knotx-adapter/knotx-adapter-common/src/main/java/io/knotx/adapter/common/http/HttpClientFacade.java

```diff
@@ -124,28 +119,28 @@ protected ClientRequest buildServiceRequest(ClientRequest originalRequest, JsonO
     return services.stream().filter(metadata -> servicePath.matches(metadata.getPath())).findAny();
   }
 
-  private Observable<HttpClientResponse> callService(
+  private Single<HttpResponse<Buffer>> callService(
       Pair<ClientRequest, ServiceMetadata> serviceData, HttpMethod method) {
+    final Single<HttpResponse<Buffer>> httpResponse;
+
     final ClientRequest serviceRequest = serviceData.getLeft();
     final ServiceMetadata serviceMetadata = serviceData.getRight();
 
-    return Observable.create(subscriber -> {
-      HttpClientRequest httpRequest = httpClient
-          .request(method, serviceMetadata.getPort(), serviceMetadata.getDomain(),
-              serviceRequest.getPath());
-      Observable<HttpClientResponse> resp = httpRequest.toObservable();
-      resp.subscribe(subscriber);
-
-      MultiMap filteredHeaders = getFilteredHeaders(serviceRequest.getHeaders(),
-          serviceMetadata.getAllowedRequestHeaderPatterns());
-      filteredHeaders.names().forEach(
-          headerName -> httpRequest.putHeader(headerName, filteredHeaders.get(headerName)));
-      if (!serviceRequest.getFormAttributes().isEmpty()) {
-        httpRequest.end(UrlEncodedBodyBuilder.encodeBody(serviceRequest.getFormAttributes()));
-      } else {
-        httpRequest.end();
-      }
-    });
+    final HttpRequest<Buffer> request = webClient
+        .request(method, serviceMetadata.getPort(), serviceMetadata.getDomain(),
+            serviceRequest.getPath());
+
+    MultiMap filteredHeaders = getFilteredHeaders(serviceRequest.getHeaders(),
+        serviceMetadata.getAllowedRequestHeaderPatterns());
+    filteredHeaders.names().forEach(
+        headerName -> request.putHeader(headerName, filteredHeaders.get(headerName)));
```

Comment:
I think we should use filteredHeaders.getAll(headerName), as the multimap can have multiple items for the given key, otherwise we might lost e.g. multiple cookies if it's a cookie header.